### ISSUE TITLE
osu!lazer display wrong rank

### ISSUE THREAD

**Issue by NamGitHub2002** [ppy/osu#25164]
### Type

Other

### Bug description

Play mode: osu!catch
My friend has a rank problem, he plays rank 42 in the rankings, but osu!lazer shows up as rank 2.
The maps are showing the wrong rank:
1. Tear Rain - cYsmix - mapped by [jonathanlfj]
https://osu.ppy.sh/beatmapsets/140662/#fruits/351190
2. Manic - cYsmix - mapped by [Monstrata]
https://osu.ppy.sh/beatmapsets/361175/#fruits/793620
3. triangles - cYsmix - mapped by [ohm002]
https://osu.ppy.sh/beatmapsets/1646678#fruits/3361072

### Screenshots or videos

![image](https://github.com/ppy/osu/assets/120299835/73b09bcd-7d72-4484-bb2b-ef1e6387b7f9)
![image](https://github.com/ppy/osu/assets/120299835/1e8d973f-8d0f-466d-b730-c4fa70a87bef)
![image](https://github.com/ppy/osu/assets/120299835/f6a75ae3-2264-4082-bb69-95120d642eb7)

### Version

2023.1008.1

### Logs

[performance.log](https://github.com/ppy/osu/files/12989874/performance.log)
[network.log](https://github.com/ppy/osu/files/12989899/network.log)
[runtime.log](https://github.com/ppy/osu/files/12989882/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12989887/updater.log)
[database.log](https://github.com/ppy/osu/files/12989892/database.log)
[input.log](https://github.com/ppy/osu/files/12989896/input.log)

**Comment by longnguyen2004** [comment]
The rank in the personal best section doesn't take Classic scores into account. I'm expecting this to be very low priority, since all lazer scores will be wiped and the 2 score systems will be unified once [this](https://github.com/orgs/ppy/projects/13) is complete.

**Comment by bdach** [comment]
Duplicate of https://github.com/ppy/osu/issues/14710

To be addressed by https://github.com/ppy/osu/issues/18878
